editoast/src/views/timetable/stdcm.rs

```diff
@@ -276,44 +294,80 @@ async fn stdcm(
         comfort: stdcm_request.comfort,
         path_items,
         start_time: earliest_departure_time,
-        trains_requirements,
+        trains_requirements: trains_requirements.clone(),
         maximum_departure_delay,
         maximum_run_time,
         speed_limit_tag: stdcm_request.speed_limit_tags,
         time_gap_before: stdcm_request.time_gap_before,
         time_gap_after: stdcm_request.time_gap_after,
         margin: stdcm_request.margin,
         time_step: Some(2000),
-        work_schedules: match stdcm_request.work_schedule_group_id {
-            Some(work_schedule_group_id) => {
-                build_work_schedules(
-                    conn,
-                    earliest_departure_time,
-                    maximum_run_time,
-                    work_schedule_group_id,
-                )
-                .await?
-            }
-            None => vec![],
-        },
+        work_schedules: build_stdcm_work_schedules(
+            &work_schedules,
+            earliest_departure_time,
+            maximum_run_time,
+        ),
         temporary_speed_limits,
+    };
+
+    let stdcm_response = stdcm_request.fetch(core_client.as_ref()).await?;
+
+    if let STDCMResponse::PathNotFound = stdcm_response {
```

Comment:
Done.

Reply:
Agreed. The ideal code presentation to me would be something in the lines of:

```rust
fn stdcm(input: StdcmInput) -> Result<StdcmOutput> {
	let res1: Step1Result = step_1(input)?;
	let res2: Step2Result = step_2(res1)?;
	if res2.success() {
		return Ok(res2.into());
	}
	let res3: FallbackResult = step_fallback(res2)?;
	Ok(res4.into())
}
```

This way each step would have its own structure with exactly what it needs and nothing else making reading easier and changes more explicit in the diffs. And it provides many definitions to document the process properly.

You are probably discussing this with @woshilapin right now, and this can of course be a topic for another PR. But I feel we're reaching an amount of code (and of features) justifying a refactoring to keeps things organized.

Reply:
See https://github.com/OpenRailAssociation/osrd/pull/9434